pkg/config/config.go

```diff
@@ -30,38 +50,125 @@ const (
 	// TODO consider moving DDSite here as well
 )
 
-// GetWatchNamespaces returns the Namespaces the operator should be watching for changes.
-func GetWatchNamespaces() []string {
-	ns, found := os.LookupEnv(WatchNamespaceEnvVar)
-	if !found {
-		return nil
+// CacheOptions function configures Controller Runtime cache options on a resource level (supported in v0.16+).
+// Datadog CRDs and additional resources required for their reconciliation will be cached only if the respective feature is enabled.
+func CacheOptions(logger logr.Logger, datadogAgentEnabled, datadogMonitorEnabled, datadogSLOEnabled, datadogAgentProfileEnabled bool) cache.Options {
+	byObject := map[client.Object]cache.ByObject{}
+
+	if datadogAgentEnabled {
+		agentNamespaces := getWatchNamespacesFromEnv(logger, agentWatchNamespaceEnvVar)
+		logger.Info("DatadogAgent Enabled", "watching namespaces", maps.Keys(agentNamespaces))
+		byObject[&datadoghqv2alpha1.DatadogAgent{}] = cache.ByObject{
+			Namespaces: agentNamespaces,
+		}
 	}
 
-	// Add support for MultiNamespace set in WATCH_NAMESPACE (e.g ns1,ns2).
-	if strings.Contains(ns, ",") {
-		return strings.Split(ns, ",")
+	if datadogMonitorEnabled {
+		monitorNamespaces := getWatchNamespacesFromEnv(logger, monitorWatchNamespaceEnvVar)
+		logger.Info("DatadogMonitor Enabled", "watching namespaces", maps.Keys(monitorNamespaces))
+		byObject[&datadoghqv1alpha1.DatadogMonitor{}] = cache.ByObject{
+			Namespaces: monitorNamespaces,
+		}
+	}
+
+	if datadogSLOEnabled {
+		sloNamespaces := getWatchNamespacesFromEnv(logger, sloWatchNamespaceEnvVar)
+		logger.Info("DatadogSLO Enabled", "watching namespaces", maps.Keys(sloNamespaces))
+		byObject[&datadoghqv1alpha1.DatadogSLO{}] = cache.ByObject{
+			Namespaces: sloNamespaces,
+		}
 	}
 
-	return []string{ns}
+	if datadogAgentProfileEnabled {
```

Comment:
should we add these cache filters if introspection is enabled?

Reply:
Yes, thanks for pointing this out. Fixed in 9th commit.